packages/oats-runtime/src/redirect.ts

```diff
@@ -0,0 +1,72 @@
+import type { Response } from './server';
+import encodeUrl = require('encodeurl');
+import escapeHtml = require('escape-html');
+
+const REDIRECT_STATUSES = [300, 301, 302, 303, 305, 307, 308] as const;
+const DEFAULT_REDIRECT_STATUS = 302;
+const TEXT_HTML_CONTENT_TYPE = 'text/html';
+const TEXT_PLAIN_CONTENT_TYPE = 'text/plain';
+
+export type RedirectStatus = typeof REDIRECT_STATUSES[number];
+
+type DefaultValue<ContentType extends string> = ContentType extends
+  | typeof TEXT_HTML_CONTENT_TYPE
+  | typeof TEXT_PLAIN_CONTENT_TYPE
+  | `${typeof TEXT_HTML_CONTENT_TYPE};${string}`
+  | `${typeof TEXT_PLAIN_CONTENT_TYPE};${string}`
+  ? string
+  : null;
+
+export function redirect<
+  Status extends RedirectStatus = typeof DEFAULT_REDIRECT_STATUS,
+  ContentType extends string = typeof TEXT_HTML_CONTENT_TYPE,
+  Value = DefaultValue<ContentType>
+>(
+  url: string,
+  options: {
+    status?: Status;
+    contentType?: ContentType;
+    value?: Value;
+  } = {}
+): Response<Status, ContentType, Value, Record<string, any>> {
```

Comment:
why headers is `Record<string, any>` instead of `{ location: string }`?

Reply:
right, I thought it didn't matter. Do we validate response headers so that they match openapi spec?

Reply:
I think we have a type for the headers and we do validate